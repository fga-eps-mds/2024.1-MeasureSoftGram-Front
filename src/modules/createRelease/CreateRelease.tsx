import React, { useEffect, useState } from 'react';
import { Alert, Container, Snackbar } from '@mui/material';

import DrawerMenu from '@components/DrawerMenu';
import { ButtonType, Product } from '@customTypes/product';
import api from '@services/api';
import { useSnackbar } from 'notistack';
import ConfigPage from './components/ConfigPage';
import ReleaseInfo from './components/ReleaseInfo';
import ReleaseGoals from './components/ReleaseGoals';

import { CreateReleaseProvider, useCreateReleaseContext } from './context/useCreateRelease';
import FirstReleaseWarning from './components/FirstReleaseWarning';
import ReleaseConfigSelector from './components/ReleaseConfigSelector';
import ThresholdConfig from './components/ThresholdConfig';
import { CREATE_RELEASE_STEP } from './consts';

interface CreateReleaseProps {
  open: boolean;
  handleClose: () => void;
}

interface CreateReleaseContainerProps {
  productId: string;
  organizationId: string;
  currentProduct: Product;
  open: boolean;
  handleClose: () => void;
}

function CreateRelease({ open, handleClose }: CreateReleaseProps) {
  const [activeStep, setActiveStep] = useState(0);
  const [configPage, setConfigPage] = useState(0);
  const [configPageTitle, setConfigPageTitle] = useState<string>('');
  const configPageTitles = [
    'Definir peso das características',
    'Definir peso das subcaracterísticas',
    'Definir peso das medidas'
  ];
  const { enqueueSnackbar } = useSnackbar()

  const {
    alertMessage,
    closeAlert,
    goToNextStep,
    finishReleasePlanning,
    configPageData,
    releaseInfoForm,
    organizationId,
    productId,
    getNextStep,
    getPreviousStep,
    isFirstRelease,
    resetStates
  } = useCreateReleaseContext();

  const renderStep = () =>
  ({
    0: <ReleaseInfo setActiveStep={setActiveStep} />,
    1: isFirstRelease ? <FirstReleaseWarning /> : <ReleaseConfigSelector setActiveStep={setActiveStep} />,
    2: (
      <ConfigPage
        setActiveStep={setActiveStep}
        page={configPage}
        title={configPageTitle}
      />
    ),
    3: (
      <ThresholdConfig
        data={configPageData.characteristicData}
        tabs={configPageData.subcharacterCheckbox}
        checkboxValues={configPageData.measureCheckbox}
        setCheckboxValues={configPageData.setMeasureCheckbox}
      />
    ),
    4: <ReleaseGoals setActiveStep={setActiveStep} />
  }[activeStep]);

  const handleCloseModal = () => {
    handleClose();
    closeAlert();
    resetStates();
    setActiveStep(0);
  };

  const handleGoToNextStep = () => {
    if (!goToNextStep(activeStep)) return;

    const newActiveStep = getNextStep(activeStep, configPage);
    if (activeStep === newActiveStep && newActiveStep === CREATE_RELEASE_STEP.CharacteristicStep) {
      setConfigPage(configPage + 1);
    }
    setActiveStep(newActiveStep);
  };

  const handleBackButton = () => {
    if (activeStep === CREATE_RELEASE_STEP.ReleaseInfoStep) {
      handleCloseModal();
      return;
    }

    const newActiveStep = getPreviousStep(activeStep, configPage);
    if (activeStep === newActiveStep && newActiveStep === CREATE_RELEASE_STEP.CharacteristicStep) {
      setConfigPage(configPage - 1);
    }
    setActiveStep(newActiveStep);
  };

  useEffect(() => {
    setConfigPageTitle(configPageTitles[configPage]);
  }, [configPage]);

  useEffect(() => {
    if (open) resetStates();
  }, [open])

  async function checkReleaseInfo() {
    try {
      const { data, status } = await api.get(`/organizations/${organizationId}/products/${productId}/release/is-valid/?nome=${releaseInfoForm.name}&dt-inicial=${releaseInfoForm.startDate}&dt-final=${releaseInfoForm.endDate}`)
      handleGoToNextStep()
    } catch (error: any) {
      console.log("error.response", error.response)
      console.log("error", error)
      enqueueSnackbar(`Erro ao criar release, ${error?.response?.data?.detail}`, { variant: 'error' })
    }
  }
  const handleNextButton = async () => {
    if (activeStep === CREATE_RELEASE_STEP.ReleaseInfoStep) {
      checkReleaseInfo();
    } else if (activeStep === CREATE_RELEASE_STEP.ReleaseGoalStep) {
      finishReleasePlanning();
    } else {
      handleGoToNextStep();
    }
  }

  const BUTTONS: Array<ButtonType> = [

    {
      label: 'Cancelar',
      onClick: handleCloseModal,
      backgroundColor: '#FFF',
      color: '#113D4C',
      variant: 'outlined',
      dataTestId: 'cancel-button'
    },
    {
      label: 'Voltar',
      onClick: handleBackButton,
      backgroundColor: '#FFF',
      color: '#113D4C',
      variant: 'outlined',
      dataTestId: 'back-button'
    },
    {
      label: activeStep === CREATE_RELEASE_STEP.ReleaseGoalStep ? 'Finalizar' : 'Continuar',
      onClick: async () => await handleNextButton(),
      backgroundColor: '#113D4C',
      color: '#fff',
      variant: 'contained',
      disabled: activeStep === CREATE_RELEASE_STEP.ReleaseSelectorStep && !isFirstRelease,
      dataTestId: 'next-button'
    }
  ];

  const ALERTS = {
    successOnCreation: <Alert severity="success">Release criada com sucesso!</Alert>,
    errorOnCreation: <Alert severity="error">Ocorreu um erro durante a criação da release.</Alert>,
    noCharacteristicSelected: <Alert severity='warning'>Ao menos uma característica deve ser selecionada</Alert>,
    fillName: <Alert severity='warning'>Preencha o nome da release</Alert>,
    characteristicValuesInvalid: <Alert severity='warning'>A soma de todos os valores preenchidos deve ser igual a 100</Alert>,
    invalidDate: <Alert severity='warning'>A data de início deve ser anterior a data de fim</Alert>,
    sameReleaseGoal: <Alert severity='warning'>As informações da release devem ser diferentes da última release</Alert>,
  };
  const handleOnCloseAlert = () => {
    if (alertMessage === 'successOnCreation') {
      handleClose();
      setActiveStep(0);
    }

    setTimeout(closeAlert, 5000);
  };

  useEffect(() => {
    if (alertMessage) handleOnCloseAlert();
  }, [alertMessage]);

  return (
    <>
      <DrawerMenu handleCloseModal={handleCloseModal} open={open} buttons={BUTTONS}>
        <Container>{renderStep()}</Container>
      </DrawerMenu>

      <Snackbar
        anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
        open={!!alertMessage}
        onClose={handleOnCloseAlert}
      >
        {ALERTS[alertMessage as keyof typeof ALERTS]}
      </Snackbar>
    </>
  );
}

function CreateReleaseContainer({
  productId,
  organizationId,
  open,
  handleClose,

  currentProduct
}: CreateReleaseContainerProps) {
  return (
    <CreateReleaseProvider productId={productId} organizationId={organizationId} currentProduct={currentProduct}>
      <CreateRelease open={open} handleClose={handleClose} />
    </CreateReleaseProvider>
  );
}

export default CreateReleaseContainer;
