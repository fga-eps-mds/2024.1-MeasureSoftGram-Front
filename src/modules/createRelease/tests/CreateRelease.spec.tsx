import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import CreateRelease from '../CreateRelease';

const product = {
  id: "1",
  name: "MeasureSoftGram",
  description: '',
  github_url: '',
  created_at: '',
  updated_at: '',
  gaugeRedLimit: 0,
  gaugeYellowLimit: 0
}

jest.mock('@modules/createRelease/context/useCreateRelease', () => ({
  ...jest.requireActual('@modules/createRelease/context/useCreateRelease'),
  useCreateReleaseContext: () => ({
    closeAlert: () => { },
    goToGoalsStep: () => { },
    resetStates: () => { },
    goToNextStep: () => true,
    finishReleasePlanning: () => { },
    releaseInfoForm: {
      characteristics: ['reliability', 'maintainability'],
      endDate: '2022-09-25',
      name: 'asdasd',
      description: 'asdasd',
      goal: 1,
      startDate: '2022-09-18'
    },
    alertMessage: 'successOnCreation',
    getNextStep: jest.fn((step: number) => step + 1),
    getPreviousStep: jest.fn((step: number) => step - 1),
    createProductReleaseGoal: () => { },
    configPageData: {
      characteristicData: [],
    }
  })
}));

describe('<CreateRelease />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(<CreateRelease open={true} handleClose={() => { }} productId="1" organizationId="1" currentProduct={product} />);

      expect(tree).toMatchSnapshot();
    });
  });

  it('Deve chamar o handleNextButton para todas as páginas', () => {
    const tree = render(<CreateRelease open={true} handleClose={() => { }} productId="1" organizationId="1" currentProduct={product} />);
    const { getByTestId } = tree;
    const nextButton = getByTestId('next-button');
    expect(nextButton).toBeDefined();
    fireEvent.click(nextButton);
    fireEvent.click(nextButton);
    fireEvent.click(nextButton);
    fireEvent.click(nextButton);
    fireEvent.click(nextButton);
    const backButton = getByTestId('back-button');
    expect(backButton).toBeDefined();
    const cancelButton = getByTestId('cancel-button');
    expect(cancelButton).toBeDefined();
    fireEvent.click(cancelButton);
    fireEvent.click(backButton);
    fireEvent.click(backButton);
  });

  it('Deve fechar o modal e redefinir os estados ao clicar em Cancelar', () => {
    const handleClose = jest.fn();
    const { getByTestId } = render(
      <CreateRelease open handleClose={handleClose} productId="1" organizationId="1" currentProduct={product} />
    );

    fireEvent.click(getByTestId('cancel-button'));

    // Verifica se a função de fechamento foi chamada
    expect(handleClose).toHaveBeenCalled();
  });

});
