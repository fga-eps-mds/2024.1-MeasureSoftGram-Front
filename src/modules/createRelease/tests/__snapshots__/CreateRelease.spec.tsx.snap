// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRelease /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomRight css-sfbrjq-MuiSnackbar-root"
        role="presentation"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
          direction="up"
          role="alert"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="SuccessOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            Release criada com sucesso!
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1u2w381-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorRight css-1160xiw-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <div
          class="MuiBox-root css-1m45ylu"
        >
          <div
            class="styles__ScrollDiv-sc-1casz44-0 hcLjsI"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
            >
              <div
                class="styles__Header-sc-1xrutlg-0 ihDdkb"
              >
                <h1
                  style="color: rgb(51, 86, 142); font-weight: 500;"
                >
                  Planejamento de Release
                </h1>
                <nav
                  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1gui7g7-MuiTypography-root-MuiBreadcrumbs-root"
                >
                  <ol
                    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                  >
                    <li
                      class="MuiBreadcrumbs-li"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-46jzwd-MuiTypography-root-MuiLink-root"
                      >
                        Criar Release
                      </a>
                    </li>
                    <li
                      aria-hidden="true"
                      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                    >
                      <span
                        class="MuiBox-root css-1bi6evy"
                      />
                    </li>
                    <li
                      class="MuiBreadcrumbs-li"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-ae5t40-MuiTypography-root-MuiLink-root"
                      >
                        Definir configuração do modelo
                      </a>
                    </li>
                    <li
                      aria-hidden="true"
                      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                    >
                      <span
                        class="MuiBox-root css-1bi6evy"
                      />
                    </li>
                    <li
                      class="MuiBreadcrumbs-li"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-ae5t40-MuiTypography-root-MuiLink-root"
                      >
                        Balancear características
                      </a>
                    </li>
                  </ol>
                </nav>
              </div>
              <div
                class="styles__Body-sc-1xrutlg-1 lojNHO"
              >
                <div
                  class="MuiBox-root css-0"
                >
                  <h2
                    style="color: rgb(83, 139, 163); font-weight: 500;"
                  >
                    Configurações básicas
                  </h2>
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    style="margin-bottom: 24px;"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for=":r0:"
                      id=":r0:-label"
                    >
                      Apelido da release
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        data-testid="apelido-release"
                        id=":r0:"
                        required=""
                        type="text"
                        value="asdasd"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Apelido da release
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    style="margin-bottom: 24px;"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for=":r1:"
                      id=":r1:-label"
                    >
                      Descrição da release
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                        data-testid="descricao-release"
                        id=":r1:"
                        style="height: 0px; overflow: hidden;"
                      >
                        asdasd
                      </textarea>
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Descrição da release
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-k008qs"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
                      style="margin-right: 16px;"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for=":r2:"
                        id=":r2:-label"
                      >
                        Data de ínicio da release
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          data-testid="inicio-release"
                          id=":r2:"
                          required=""
                          type="date"
                          value="2022-09-18"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-14lo706"
                          >
                            <span>
                              Data de ínicio da release
                               
                              *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for=":r3:"
                        id=":r3:-label"
                      >
                        Data de fim da release
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          data-testid="fim-release"
                          id=":r3:"
                          required=""
                          type="date"
                          value="2022-09-25"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-14lo706"
                          >
                            <span>
                              Data de fim da release
                               
                              *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1iy6m51-MuiButtonBase-root-MuiButton-root"
              data-testid="cancel-button"
              tabindex="0"
              type="button"
            >
              Cancelar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1iy6m51-MuiButtonBase-root-MuiButton-root"
              data-testid="back-button"
              tabindex="0"
              type="button"
            >
              Voltar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-8jrlok-MuiButtonBase-root-MuiButton-root"
              data-testid="next-button"
              tabindex="0"
              type="button"
            >
              Continuar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomRight css-sfbrjq-MuiSnackbar-root"
      role="presentation"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="SuccessOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
        >
          Release criada com sucesso!
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
